sklearn/preprocessing/tests/test_label.py

```diff
@@ -118,15 +118,19 @@ def test_label_binarizer_set_label_encoding():
 
 
 @pytest.mark.parametrize("dtype", ["Int64", "Float64", "boolean"])
-def test_label_binarizer_pandas_nullable(dtype):
+@pytest.mark.parametrize("unique_first", [True, False])
+def test_label_binarizer_pandas_nullable(dtype, unique_first):
     """Checks that LabelBinarizer works with pandas nullable dtypes.
 
     Non-regression test for gh-25637.
     """
     pd = pytest.importorskip("pandas")
-    from sklearn.preprocessing import LabelBinarizer
 
     y_true = pd.Series([1, 0, 0, 1, 0, 1, 1, 0, 1], dtype=dtype)
+    if unique_first:
+        # Calling unique creates a pandas array which has a different interface
+        # compared to a pandas Series. Specifically, pandas arrays do not have "iloc".
+        y_true = y_true.unique()
```

Comment:
I understand that the original issue is about labelbinarizer, but it's actually a fix of check_array so I think we need a test in ``test_validation``.